packages/ra-core/src/form/FormDataConsumer.tsx

```diff
@@ -117,8 +119,8 @@ export const FormDataConsumerView: SFC<Props> = ({
     return ret === undefined ? null : ret;
 };
 
-const FormDataConsumer = (props: ConnectedProps) => {
-    const formState = useFormState();
+const FormDataConsumer = ({ subscription, ...props }: ConnectedProps) => {
+    const formState = useFormState(subscription ? { subscription } : undefined);
```

Comment:
I think you don't need the test. Passing `useFormState({ subscription })` will default to all subscriptions if `subscription` is undefined.

Reply:
ok